src/lib/core/TLVWriter.cpp

```diff
@@ -97,6 +97,22 @@ CHIP_ERROR TLVWriter::Finalize()
     return err;
 }
 
+CHIP_ERROR TLVWriter::ReserveBuffer(uint32_t aBufferSize)
+{
+    VerifyOrReturnError(mRemainingLen >= aBufferSize, CHIP_ERROR_NO_MEMORY);
+
+    uint32_t spaceLeftInCurrentBuffer        = mReservedSize + mRemainingLen;
```

Comment:
This logic is not easy to follow, and it's unclear to me why the IsSafeToReserve check doesn't depend on aBufferSize at all?

Reply:
So if I'm understanding this correctly, reserving with a backing store is allowed if there is space in the current buffer AND the max length of the writer happens to have been set to a value less than or equal to the first buffer the backing store happens to have provided? This seems a little difficult for clients if this API to predict.

Reply:
I guess I didn't want to limit the restrictions I was putting in on some future `TLVBackingStore` that might be more capable. I realize that this shouldn't be at all the focus of this PR I would remove line along with
```
    bool canPossiblyAllocateAdditionalMemory = spaceLeftInCurrentBuffer < mMaxLen;
    if (mBackingStore && canPossiblyAllocateAdditionalMemory)
    {
        VerifyOrReturnError(mBackingStore->IsSafeToReserve(), CHIP_ERROR_INCORRECT_STATE);
    }
```